src/metabase/driver.clj

```diff
@@ -585,7 +585,11 @@
 
     ;; Does the driver support a faster `sync-fields` step by fetching all FK metadata in a single collection?
     ;; if so, `metabase.driver/describe-fields` must be implemented instead of `metabase.driver/describe-table`
-    :describe-fields})
+    :describe-fields
+
+    ;; Does the driver support automatically adding an auto-incrementing integer column for uploads?
+    ;; DEFAULTS TO TRUE
+    :auto-incrementing-upload-column})
```

Comment:
the name confuses me a bit, how about "upload-with-auto-pk"?

Reply:
and we should mention this in the driver change log.

Reply:
1. `upload-with-auto-pk` is nice. 
2. `create-auto-pk-with-append-csv?` was a driver method for no particular reason. I think it was a mistake to make it a driver method given we can use features for booleans and with less code, so I'm using a feature for this now.
3. yes, good point

Reply:
updated cf928b0